pyqtgraph/graphicsItems/NonUniformImage.py

```diff
@@ -72,62 +80,87 @@ def getHistogram(self, **kwds):
 
         return hist[1][:-1], hist[0]
 
-    def generatePicture(self):
+    def setLevels(self, levels):
+        self.levels = levels
+        self.picture = None
+        self.update()
 
-        x, y, z = self.data
+    def getLevels(self):
```

Comment:
Could this be renamed to just `levels()` (with `setLevels()` being the setter)?

Reply:
ColorBarItem is looking for getLevels. To change to "levels()" would require corresponding changes to ColorBarItem, ImageItem, PColorMeshItem, HistogramLUTItem.

Reply:
Outside the scope of this PR to change all those other objects, but I'd like to deprecate that at point; but that's a future issue 👍